tests/test_controller.py

```diff
@@ -160,6 +160,34 @@ def test_is_cachable_for_no_store():
     assert not controller.is_cachable(request=request, response=response)
 
 
+def test_is_cachable_for_shared_cache():
+    controller = Controller(cache_private=False, allow_heuristics=True)
+
+    request = Request(b"GET", b"https://example.com", headers=[])
+
+    response = Response(200, headers=[(b"Cache-Control", b"public")])
+
+    assert controller.is_cachable(request=request, response=response)
+
+    response = Response(200, headers=[(b"Cache-Control", b"private")])
+
+    assert not controller.is_cachable(request=request, response=response)
+
+    response = Response(200, headers=[(b"Cache-Control", b"private=set-cookie")])
+
+    assert not controller.is_cachable(request=request, response=response)
```

Comment:
I was unsure if I should make different test cases for this (maybe even using `pytest.mark.parametrize`), but saw that there were some other tests using multiple asserts. So I did the same.

Reply:
It's okay, I like it